lib/wallaby/experimental/chrome/chromedriver/server.ex

```diff
@@ -0,0 +1,153 @@
+defmodule Wallaby.Experimental.Chrome.Chromedriver.Server do
+  @moduledoc false
+  use GenServer
+
+  alias Wallaby.Driver.Utils
+  alias Wallaby.Experimental.Chrome.Chromedriver.ReadinessChecker
+
+  defmodule State do
+    @moduledoc false
+    defstruct [:port_number, :chromedriver_path, ready?: false, calls_awaiting_readiness: []]
+
+    @type port_number :: non_neg_integer
+
+    @type t :: %__MODULE__{
+            port_number: port_number,
+            chromedriver_path: String.t(),
+            ready?: boolean(),
+            calls_awaiting_readiness: [GenServer.from()]
+          }
+  end
+
+  @type server :: GenServer.server()
+  @typep port_number :: non_neg_integer()
+
+  @default_startup_timeout :timer.seconds(10)
+
+  def child_spec(args) do
+    %{id: __MODULE__, start: {__MODULE__, :start_link, args}}
+  end
+
+  @type start_link_opt :: {:startup_timeout, timeout()} | GenServer.option()
+
+  @spec start_link(String.t(), [start_link_opt]) :: GenServer.on_start()
+  def start_link(chromedriver_path, opts \\ [])
+      when is_binary(chromedriver_path) and is_list(opts) do
+    {start_opts, opts} = Keyword.split(opts, [:name])
+    GenServer.start_link(__MODULE__, {chromedriver_path, opts}, start_opts)
+  end
+
+  @spec get_base_url(server) :: String.t()
+  def get_base_url(server) do
+    server
+    |> GenServer.call(:get_port_number)
+    |> build_base_url()
+  end
+
+  @spec wait_until_ready(server, timeout()) :: :ok | {:error, :timeout}
+  def wait_until_ready(server, timeout \\ 5000) do
+    GenServer.call(server, :wait_until_ready, timeout)
+  catch
+    :exit, {:timeout, _} ->
+      {:error, :timeout}
+  end
+
+  @impl true
+  def init({chromedriver_path, opts}) do
+    port_number = Utils.find_available_port()
+
+    startup_timeout = Keyword.get(opts, :startup_timeout, @default_startup_timeout)
+    Process.send_after(self(), :ensure_readiness, startup_timeout)
+
+    send(self(), :start_chromedriver)
```

Comment:
This might be a good place for `handle_continue`

Reply:
I was wondering if I could use `handle_continue` since it requires OTP 21+. What is the minimum OTP version we're require for wallaby? I saw Elixir 1.7 supports back to OTP 19. I'm totally happy to change it if it we are only supporting OTP 21+.

Reply:
I think we are safe to only support OTP 21. In #447 I also made that trade-off to be able to use `__STACKTRACE__`.